### PR TITLE
Recreate plugin with correct name

### PR DESCRIPTION
**Pull request by SimoneBressan**

### CODE REVIEW

File: android/src/main/kotlin/net/flutter_location_listener/kuama/flutter_location_listener/FlutterLocationListenerPlugin.kt [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+import io.flutter.embedding.engine.loader.FlutterLoader
+import io.flutter.embedding.engine.plugins.FlutterPlugin
+import io.flutter.plugin.common.BinaryMessenger
+import io.flutter.plugin.common.MethodCall
+import io.flutter.plugin.common.MethodChannel
+import io.flutter.view.FlutterCallbackInformation
+
+import net.kuama.android.backgroundLocation.service.BackgroundService
+import java.lang.IllegalArgumentException
+
```
Comment (kuamanet) [inline android/src/main/kotlin/net/flutter_location_listener/kuama/flutter_location_listener/FlutterLocationListenerPlugin.kt:17]:
```suggestion
  private const val SHARED_PREFERENCES_KEY = "geofencing_plugin_cache"
  private const val USER_CALLBACK_ID_KEY = "userCallbackId"
  private const val PLUGIN_CALLBACK_ID_KEY = "pluginCallbackId"
```

File: android/src/main/kotlin/net/flutter_location_listener/kuama/flutter_location_listener/FlutterLocationListenerPlugin.kt [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+/** FlutterLocationListenerPlugin */
+class FlutterLocationListenerPlugin: FlutterPlugin, MethodChannel.MethodCallHandler {
+  private val SHARED_PREFERENCES_KEY = "geofencing_plugin_cache"
+  private val USER_CALLBACK_ID_KEY = "userCallbackId"
+  private val PLUGIN_CALLBACK_ID_KEY = "pluginCallbackId"
```
Comment (kuamanet) [inline android/src/main/kotlin/net/flutter_location_listener/kuama/flutter_location_listener/FlutterLocationListenerPlugin.kt]:
```suggestion

```

File: android/src/main/kotlin/net/flutter_location_listener/kuama/flutter_location_listener/FlutterLocationListenerPlugin.kt [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+        context.registerReceiver(receiver, IntentFilter(BackgroundService::class.java.name))
+        context.startService(Intent(context, BackgroundService::class.java))
+        result.success(null)
+      }
+      "currentLocation" -> {
+        result.success(location?.toMap())
+      }
+      "stopService" -> {
+        result.success(null)
```
Comment (kuamanet) [inline android/src/main/kotlin/net/flutter_location_listener/kuama/flutter_location_listener/FlutterLocationListenerPlugin.kt:67]:
```suggestion
        stopService(Intent(context, BackgroundService::class.java))
        result.success(null)
```

File: android/src/main/kotlin/net/flutter_location_listener/kuama/flutter_location_listener/FlutterLocationListenerPlugin.kt [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+      }
+    }
+  }
+
+  private var backgroundMethodChannel: MethodChannel? = null
+
+  private inner class MyBroadcastReceiver: BroadcastReceiver() {
```
Comment (kuamanet) [inline android/src/main/kotlin/net/flutter_location_listener/kuama/flutter_location_listener/FlutterLocationListenerPlugin.kt]:
```suggestion
  private inner class LocationBroadcastReceiver: BroadcastReceiver() {
```

File: android/src/main/kotlin/net/flutter_location_listener/kuama/flutter_location_listener/FlutterLocationListenerPlugin.kt [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+        val args = DartExecutor.DartCallback(context.assets, flutterLoader.findAppBundlePath(), userCallbackInformation)
+
+        val engine = FlutterEngine(context)
+        engine.dartExecutor.executeDartCallback(args)
+
+        backgroundMethodChannel = MethodChannel(engine.dartExecutor, "flutter_location_listener#callback")
+      }
+
+      val paramsMap = HashMap<String, Any>()
```
Comment (kuamanet) [inline android/src/main/kotlin/net/flutter_location_listener/kuama/flutter_location_listener/FlutterLocationListenerPlugin.kt]:
```suggestion
      val paramsMap = mutableMapOf<String, Any>()
```

File: android/src/main/kotlin/net/flutter_location_listener/kuama/flutter_location_listener/FlutterLocationListenerPlugin.kt [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+}
+
+class Location constructor(private val latitude: Double, private val longitude: Double) {
+  fun toMap(): HashMap<String, Double> {
+    val hashMap: HashMap<String, Double> = HashMap()
+    hashMap["latitude"] = latitude
+    hashMap["longitude"] = longitude
+    return hashMap
+  }
+}
```
Comment (kuamanet) [inline android/src/main/kotlin/net/flutter_location_listener/kuama/flutter_location_listener/FlutterLocationListenerPlugin.kt]:
```suggestion
class Location (private val latitude: Double, private val longitude: Double) {
    fun toMap(): Map<String, Double> = mapOf(
        "latitude" to latitude,
        "longitude" to longitude
    )
}
```

File: pubspec.yaml [hunk 7]
```diff
@@ -27,10 +28,10 @@ flutter:
   plugin:
     platforms:
       android:
-        package: com.example.flutter_position_tracker
-        pluginClass: FlutterPositionTrackerPlugin
+        package: net.flutter_location_listener.kuama.flutter_location_listener
```
Comment (kuamanet) [inline pubspec.yaml:31]:
```suggestion
        package: net.kuama.flutter_location_listener
```

File: test/flutter_location_listener_test.dart [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+//   });
+//
+//   tearDown(() {
+//     channel.setMockMethodCallHandler(null);
+//   });
+//
+//   test('getPlatformVersion', () async {
+//     expect(await FlutterLocationListener.platformVersion, '42');
+//   });
+// }
```
Comment (kuamanet) [inline test/flutter_location_listener_test.dart:23]: can we delete this file?

**Review by kuamanet — CHANGES_REQUESTED** [review]

File: test/flutter_location_listener_test.dart (hunk 8 above)
Comment (SimoneBressan) [reply test/flutter_location_listener_test.dart:23]: It is useful for when we go to test the plugin

**Review by kuamanet — APPROVED** [review]

**Pull request merged**